packages/cli/src/ui/hooks/usePermissionsModifyTrust.ts

```diff
@@ -45,11 +56,16 @@ function getInitialTrustState(
 export const usePermissionsModifyTrust = (
   onExit: () => void,
   addItem: UseHistoryManagerReturn['addItem'],
+  targetDirectory: string,
 ) => {
   const settings = useSettings();
-  const cwd = process.cwd();
+  const cwd = targetDirectory;
+  const isCurrentWorkspace =
+    path.resolve(targetDirectory) === path.resolve(process.cwd());
```

Comment:
This may fail on case-insensitive file systems (macOS/Windows) if the casing provided by the user differs from process.cwd(). Consider normalizing the paths (e.g., converting to lowercase) before comparison on these platforms, or using a more robust path comparison method.

Reply:
Done, converted to lowercase